Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/ExternalDB/ByExternalTable.tsx

```diff
@@ -0,0 +1,143 @@
+const emptyRecord = { ID: -1, TableName: '', ExtDBID: -1, Query: ''};
+
+const ByExternalTable: Application.Types.iByComponent = (props) => {
+    let history = useHistory();
+    const dispatch = useAppDispatch();
+
+    const data = useAppSelector(ExternalDBTablesSlice.SearchResults);
+    const status = useAppSelector(ExternalDBTablesSlice.SearchStatus);
+    const search = useAppSelector(ExternalDBTablesSlice.SearchFilters);
+    const sortField = useAppSelector(ExternalDBTablesSlice.SortField);
+    const ascending = useAppSelector(ExternalDBTablesSlice.Ascending);
+    const parentID = useAppSelector(ExternalDBTablesSlice.ParentID);
+
+    const [showNew, setShowNew] = React.useState<boolean>(false);
+    const [errors, setErrors] = React.useState<string[]>([]);
+
+    const [record, setRecord] = React.useState<SystemCenter.Types.extDBTables>(emptyRecord);
+
+    React.useEffect(() => {
+        if (parentID != null)
+            dispatch(ExternalDBTablesSlice.Fetch());
+    }, [parentID]);
+
+    React.useEffect(() => {
+        if ((status === 'unintiated' || status === 'changed') && parentID == null)
+            dispatch(ExternalDBTablesSlice.DBSearch({ filter: search }));
+    }, [status, parentID]);
+
+    function handleSelect(item) {
+        history.push({ pathname: homePath + 'index.cshtml', search: '?name=ExternalDBTable&ID=' + item.row.ID })
+    }
+
+    return (
+        <div style={{ width: '100%', height: '100%' }}>
+            <SearchBar<SystemCenter.Types.extDBTables>
+                CollumnList={ExternalDBSearchField}
+                SetFilter={(flds) => dispatch(ExternalDBTablesSlice.DBSearch({ filter: flds }))}
+                Direction={'left'}
+                defaultCollumn={ExternalDBDefaultSearchField}
+                Width={'50%'}
+                Label={'Search'}
+                ShowLoading={status == 'loading'}
+                ResultNote={status == 'error' ? 'Could not complete Search' : 'Found ' + data.length + ' External Table(s)'}
+            >
+                <li className="nav-item" style={{ width: '15%', paddingRight: 10 }}>
+                    <fieldset className="border" style={{ padding: '10px', height: '100%' }}>
+                        <legend className="w-auto" style={{ fontSize: 'large' }}>Actions:</legend>
+                        <form>
+                            <button className="btn btn-primary" hidden={props.Roles.indexOf('Administrator') < 0 && props.Roles.indexOf('Transmission SME') < 0} onClick={(event) => {
+                                event.preventDefault()
+                                setRecord({ ...emptyRecord });
+                                setShowNew(true);
+                            }}>Add External Table</button>
+                        </form>
+                    </fieldset>
+                </li>
+            </SearchBar>
+
+            <div style={{ width: '100%', height: 'calc( 100% - 136px)' }}>
```

Comment:
this is going to cause problems. We have no guarantee that the searchbar is 136px. wrap the searchbar in a row (if it is not already) and put this in a row and col